# HG changeset patch
# User Rik <rik@octave.org>
# Date 1515718033 28800
#      Thu Jan 11 16:47:13 2018 -0800
# Node ID c12b5cec5c6c57c84206af7a4f37d255ed4ebd37
# Parent  6ece7c7e4912211a9fe1846edd29fbd9520041fd
contour3.m: Pull axes configuration commands inside unwind_protect block.

* contour3.m: Pull axes configuration commands inside unwind_protect block
as all other draw routines do.

diff --git a/scripts/plot/draw/contour3.m b/scripts/plot/draw/contour3.m
--- a/scripts/plot/draw/contour3.m
+++ b/scripts/plot/draw/contour3.m
@@ -69,27 +69,28 @@ function [c, h] = contour3 (varargin)
   oldfig = [];
   if (! isempty (hax))
     oldfig = get (0, "currentfigure");
   endif
   unwind_protect
     hax = newplot (hax);
 
     [ctmp, htmp] = __contour__ (hax, "auto", varargin{:});
+
+  if (! ishold ())
+    set (hax, "view", [-37.5, 30], "box", "on",
+              "xgrid", "on", "ygrid", "on", "zgrid", "on");
+  endif
+
   unwind_protect_cleanup
     if (! isempty (oldfig))
       set (0, "currentfigure", oldfig);
     endif
   end_unwind_protect
 
-  if (! ishold ())
-    set (hax, "view", [-37.5, 30],
-              "xgrid", "on", "ygrid", "on", "zgrid", "on");
-  endif
-
   if (nargout > 0)
     c = ctmp;
     h = htmp;
   endif
 
 endfunction
 
 
